### PR TITLE
Update to 2024

### PR DESCRIPTION
**Pull request by marks-lolcode**
Updated event info to 2024
Updated Neon URLs to match new production URLs
Added Registration Hold feature
Added handling for Day Passes
Corrected Names of contributors
Incremented version to 2024.1

### CODE REVIEW

File: js/options.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ function save_options()
     var mngtOverride = document.getElementById( 'mngtoverride' ).checked;
     //var websocketServer = document.getElementById('websocketserver').value;
 
-    var pass1 = "password-goes-here";    //TODO Encode or encrypt this somehow
+    //This password is used to allow Help Desk to allow badge printing even if the status is Red
+    var pass1 = "Connie4Life";    //TODO Encode or encrypt this somehow
```
Comment (rkaiser-dev) [inline js/options.js:10]: Is this the actual password? We should remove it (and change it, if so).

**Review by rkaiser-dev — COMMENTED** [review]
I saw also that the `.pem` had been checked in and later removed. Is there a process to rotate that key?

**Comment by marks-lolcode** [comment]
Yeah, I didn't realize it had the key in it so I pulled it out when I saw
it. I have asked the CTO where we should store keys. I have no idea about
rotating them.

All this password does is allow a badge to be printed for someone who came
back red (registration hold, ops hold, etc.). The risk of having this
exposed is nothing. Someone would need to do some elaborate spy plotting to
make any use of this.

That said, I would love (as the note says) if we could encode this somehow
(maybe store a hash and compare what we type in the box to the hash?).

**Comment by rkaiser-dev** [comment]
> Yeah, I didn't realize it had the key in it so I pulled it out when I saw it. I have asked the CTO where we should store keys. I have no idea about rotating them.

No worries. I can check into it to see what we can do regarding the `.pem`.

> All this password does is allow a badge to be printed for someone who came back red (registration hold, ops hold, etc.). The risk of having this exposed is nothing. Someone would need to do some elaborate spy plotting to make any use of this. That said, I would love (as the note says) if we could encode this somehow (maybe store a hash and compare what we type in the box to the hash?).

I would also have to look into options here since I'm not as familiar with Chrome extensions. If we're looking to obfuscate securely we're looking at maybe hash+salt here

**Comment by rkaiser-dev** [comment]
FWIW, btw, I would consider that `.pem` as no longer valid, I will make sure to prioritize some time to figure out what we can do about it.

**Pull request merged**